connection.go

```diff
@@ -101,9 +101,14 @@ func (p *Pool) Exec(req *Request) ([]byte, error) {
 	conn := p.Get()
 	defer p.Put(conn)
 start:
-	if err := websocket.Message.Send(conn.ws, requestMessage); err != nil {
-		p.logger.Log("sendMessageErr", err)
-		return nil, err
+	err = websocket.Message.Send(conn.ws, requestMessage)
+	if err != nil {
```

Comment:
What about errors like `Script Evaluation Failed`?
Do they not show up at this layer?

Reply:
yes
```
{"caller":"connection.go:124","readResponseErr":"Script Evaluation Error"}
```

Reply:
So should we filter for some specific codes here rather than `err != nil`?

Reply:
oh sorry, it will show up here
```
if err != nil {
		if err == io.EOF { // EOF err we are getting on connection loss
			p.logger.Log("connectionLossErr", err)
			conn.ws, err = p.createSocket()
			if err != nil {
				return nil, err
			}
			p.logger.Log("socketRecovered", conn.id)
			goto start
		} else {
			p.logger.Log("readResponseErr", err)
			return nil, err
		}
```

Where we do filter for a specific error code